services/device-registry-base/src/main/java/org/eclipse/hono/service/management/BaseDto.java

```diff
@@ -0,0 +1,207 @@
+     * @param supplier A DTO subclass' constructor of which a new instance shall be created.
+     * @param data The data of the DTO.
+     * @param created The instant when the object was created.
+     * @param updated The instant of the most recent update.
+     * @param version The version of the DTO
+     *
+     * @param <P> The type of the DTO's payload.
+     * @param <T> The type of the DTO subclass.
+     *
+     * @return A DTO instance for updating an entry.
+     */
+    public static <P, T extends BaseDto<P>> T forRead(final Supplier<T> supplier, final P data, final Instant created, final Instant updated, final String version) {
+        final T dto = supplier.get();
+        dto.setCreationTime(created);
+        dto.setUpdatedOn(updated);
+        dto.setData(data);
+        dto.setVersion(version);
+
+        return dto;
+    }
+
+    /**
+     * Gets the version of the document.
+     *
+     * @return The version of the document.
+     */
+    public final String getVersion() {
+        return version;
+    }
+
+    /**
+     * Sets the version of the document.
+     *
+     * @param version The version of the document or {@code null} if not set.
+     */
+    protected final void setVersion(final String version) {
+        this.version = version;
+    }
+
+    /**
+     * Gets the date and time of last modification.
+     *
+     * @return The date and time of last modification.
+     */
+    public final Instant getUpdatedOn() {
+        return updatedOn;
+    }
+
+    /**
+     * Sets the date and time of last modification.
+     *
+     * @param updatedOn The date and time of last modification.
+     */
+    protected final void setUpdatedOn(final Instant updatedOn) {
+        this.updatedOn = updatedOn;
+    }
+
+    /**
+     * Gets the date and time when the entity detailed by this status was created.
+     *
+     * @return The entity's creation time.
+     */
+    public Instant getCreationTime() {
+        return creationTime;
+    }
+
+    /**
+     * Sets the date and time when the entity detailed by this status was created.
+     *
+     * @param creationTime The entity's creation time.
+     */
+    protected void setCreationTime(final Instant creationTime) {
+        this.creationTime = creationTime;
+    }
+
+    /**
+     * Gets the data object (i.e. what should be editable by clients) of this DTO.
+     * This method may be overridden by subclasses to set the JSON property name to a more meaningful one.
+     *
+     * @return The data object.
+     */
+    @JsonProperty(FIELD_DATA)
+    public T getData() {
+        return data;
+    }
+
+    /**
+     * Sets the data object of this DTO.
+     *
+     * @param data The data object.
+     */
+    protected void setData(final T data) {
+        this.data = data;
+    }
+
+    /**
+     * Gets the name of the user who did the most recent modification.
+     *
+     * @return The user's name.
+     */
+    public String getLastUser() {
+        return null;
```

Comment:
Actually, yes. I will add a comment that it is not implemented yet.

The "last-user" property actually is/was already defined (in the Device Registry Management API Specification and in the `Status` class) but I decided not to implement this in the scope of this PR since I would need to adapt even more classes adding the authenticated user which is performing the change...

Reply:
Ok, I see.